MapboxCoreNavigation/RouteController.swift

```diff
@@ -586,13 +611,15 @@ extension RouteController: CLLocationManagerDelegate {
                 RouteControllerNotificationUserInfoKey.locationKey: self.location!, //guaranteed value
                 RouteControllerNotificationUserInfoKey.rawLocationKey: location //raw
                 ])
+            // Check for a tunnel intersection whenever the current route step progresses.
+            checkForTunnelIntersection(at: location, for: manager)
         }
 
         updateDistanceToIntersection(from: location)
         updateRouteStepProgress(for: location)
         updateRouteLegProgress(for: location)
 
-        guard userIsOnRoute(location) || !(delegate?.routeController?(self, shouldRerouteFrom: location) ?? true) else {
+        guard tunnelAnimationEnabled || userIsOnRoute(location) || !(delegate?.routeController?(self, shouldRerouteFrom: location) ?? true) else {
```

Comment:
I'm slightly concerned with this. If the tunnel animation taking place, and the user is obviously outside of the tunnel, someplace else, we won't reroute them.